core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala

```diff
@@ -189,30 +189,23 @@ class BarrierTaskContextSuite extends SparkFunSuite with LocalSparkContext with
 
   test("throw exception if the number of barrier() calls are not the same on every task") {
     initLocalClusterSparkContext()
-    sc.conf.set("spark.barrier.sync.timeout", "1")
+    sc.conf.set("spark.barrier.sync.timeout", "3")
     val rdd = sc.makeRDD(1 to 10, 4)
     val rdd2 = rdd.barrier().mapPartitions { it =>
       val context = BarrierTaskContext.get()
-      try {
-        if (context.taskAttemptId == 0) {
```

Comment:
We still need to ensure that, the number of `barrier()` calls are different on all the tasks.

Reply:
I think it stills ensures, though task 0 has no `barrier()` call.